akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -368,11 +401,18 @@ private[remote] class Association(
       try {
         val outboundEnvelope = createOutboundEnvelope()
         message match {
+          case d: DeathWatchNotification if deathWatchNotificationFlushEnabled && shouldSendDeathWatchNotification(d) =>
```

Comment:
i guess this means system. messages will be re-ordered but i don't think that matters

Reply:
For different sender-receiver pairs it never matters. DeathWatchNotification is always the last message anyway, as far as I can see.